### PR TITLE
chore: Migrates `federated_settings_org_rolemapping` to the new auto-generated SDK

### PR DESCRIPTION
**Pull request by EspenAlbert**
## Description

Migrates `federated_settings_org_rolemapping` to the new auto-generated SDK

Link to any related issue(s): CLOUDP-246324

~Waiting on #2258~

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

## Further comments
Acceptance test:

```
go test -timeout 2400s -run ^TestAccFederatedSettingsOrgRoleMapping_basic$ github.com/mongodb/terraform-provider-mongodbatlas/internal/service/federatedsettingsorgrolemapping -v

=== RUN   TestAccFederatedSettingsOrgRoleMapping_basic
=== PAUSE TestAccFederatedSettingsOrgRoleMapping_basic
=== CONT  TestAccFederatedSettingsOrgRoleMapping_basic
--- PASS: TestAccFederatedSettingsOrgRoleMapping_basic (16.72s)
PASS
ok      github.com/mongodb/terraform-provider-mongodbatlas/internal/service/federatedsettingsorgrolemapping     17.835s
```

### CODE REVIEW

**Review by lmkerbey-mdb — APPROVED** [review]
LGTM.

File: internal/service/federatedsettingsorgrolemapping/data_source_federated_settings_org_role_mappings.go [hunk 1]
```diff
@@ -105,17 +98,18 @@ func dataSourceMongoDBAtlasFederatedSettingsOrganizationRoleMappingsRead(ctx con
 	return nil
 }
 
-func flattenFederatedSettingsOrganizationRoleMappings(federatedSettingsOrganizationRoleMapping *matlas.FederatedSettingsOrganizationRoleMappings) []map[string]any {
+func flattenFederatedSettingsOrganizationRoleMappings(federatedSettingsOrganizationRoleMapping *admin.PaginatedRoleMapping) []map[string]any {
```
Comment (lantoli) [inline internal/service/federatedsettingsorgrolemapping/data_source_federated_settings_org_role_mappings.go:101]: knit: I think the function can be simpliflied, e.g:

func flattenFederatedSettingsOrganizationRoleMappings(federatedSettingsOrganizationRoleMapping *admin.PaginatedRoleMapping) []map[string]any {
	var results []map[string]any
	mappings := federatedSettingsOrganizationRoleMapping.GetResults()
	for i := range mappings {
		mapping := &mappings[i]
		results = append(results, map[string]any{
			"external_group_name": mapping.GetExternalGroupName(),
			"id":                  mapping.GetId(),
			"role_assignments":    FlattenRoleAssignments(mapping.GetRoleAssignments()),
		})
	}
	return results
}

File: internal/service/federatedsettingsorgrolemapping/resource_federated_settings_org_role_mapping_test.go [hunk 2]
```diff
@@ -111,12 +117,12 @@ func importStateIDFunc(resourceName string) resource.ImportStateIdFunc {
 	}
 }
 
-func configBasic(federationSettingsID, orgID, groupID string) string {
+func configBasic(federationSettingsID, orgID, groupID, externalGroupName string) string {
 	return fmt.Sprintf(`
 	resource "mongodbatlas_federated_settings_org_role_mapping" "test" {
 		federation_settings_id = "%[1]s"
```
Comment (lantoli) [inline internal/service/federatedsettingsorgrolemapping/resource_federated_settings_org_role_mapping_test.go]: "%[1]s" can be %[1]q , etc.

**Review by lantoli — APPROVED** [review]

File: internal/service/federatedsettingsorgrolemapping/data_source_federated_settings_org_role_mappings.go (hunk 1 above)
Comment (EspenAlbert) [reply internal/service/federatedsettingsorgrolemapping/data_source_federated_settings_org_role_mappings.go:101]: Thanks. Applied in 41ee44e9ce9201e57951d72795c3949ddd31b83e ✅

File: internal/service/federatedsettingsorgrolemapping/resource_federated_settings_org_role_mapping_test.go (hunk 2 above)
Comment (EspenAlbert) [reply internal/service/federatedsettingsorgrolemapping/resource_federated_settings_org_role_mapping_test.go]: Thanks. Applied in 41ee44e9ce9201e57951d72795c3949ddd31b83e ✅

File: internal/service/federatedsettingsorgrolemapping/data_source_federated_settings_org_role_mapping.go [hunk 3]
```diff
@@ -80,20 +79,20 @@ func dataSourceMongoDBAtlasFederatedSettingsOrganizationRoleMappingRead(ctx cont
 		return diag.FromErr(errors.New("role_mapping_id must be configured"))
 	}
 
-	federatedSettingsOrganizationRoleMapping, _, err := conn.FederatedSettings.GetRoleMapping(ctx, federationSettingsID.(string), orgID.(string), roleMappingID.(string))
+	federatedSettingsOrganizationRoleMapping, _, err := conn.FederatedAuthenticationApi.GetRoleMapping(ctx, federationSettingsID.(string), roleMappingID.(string), orgID.(string)).Execute()
```
Comment (marcosuma) [inline internal/service/federatedsettingsorgrolemapping/data_source_federated_settings_org_role_mapping.go:82]: feel free to resolve after reading. I am seeing that orgID and roleMappingID are inverted, just making sure it's not undesired

**Review by marcosuma — APPROVED** [review]

File: internal/service/federatedsettingsorgrolemapping/model_federated_settings_org_role_mapping.go [hunk 4]
```diff
@@ -37,12 +39,71 @@ func FlattenRoleAssignments(roleAssignments []*matlas.RoleAssignments) []map[str
...
+						GroupId: conversion.StringPtr(roleMap["group_id"].(string)),
+						Role:    conversion.StringPtr(role.(string)),
+					}
+					roleAssignments = append(roleAssignments, roleAssignment)
+				}
+			}
+		}
+	}
+
+	sort.Sort(mRoleAssignment(roleAssignments))
```
Comment (lantoli) [inline internal/service/federatedsettingsorgrolemapping/model_federated_settings_org_role_mapping.go:72]: knit: i wouldn't have these empty lines or in next func

**Review by lantoli — APPROVED** [review]

File: internal/service/federatedsettingsorgrolemapping/data_source_federated_settings_org_role_mapping.go (hunk 3 above)
Comment (EspenAlbert) [reply internal/service/federatedsettingsorgrolemapping/data_source_federated_settings_org_role_mapping.go:82]: Correct, new signature:
`GetRoleMapping(ctx context.Context, federationSettingsId string, id string, orgId string) GetRoleMappingApiRequest`
[Github](https://github.com/mongodb/atlas-sdk-go/blob/main/admin/api_federated_authentication.go#L246)

File: internal/service/federatedsettingsorgrolemapping/model_federated_settings_org_role_mapping.go (hunk 4 above)
Comment (EspenAlbert) [reply internal/service/federatedsettingsorgrolemapping/model_federated_settings_org_role_mapping.go:72]: Removed some empty lines in f605df2

**Pull request merged**